src/domain/users/ip-metadata-validator.ts

```diff
@@ -0,0 +1,33 @@
+import { InvalidIpMetadataTypeError } from "@domain/errors"
+import { getBlockListedASNs, getAllowListedCountries } from "@config"
+
+export const IpMetadataValidator = (): IpMetadataValidator => {
+  const validate = (userIP: UserIPs): true | ValidationError => {
+    const blockListedASNs = getBlockListedASNs()
+    const allowListedCountries = getAllowListedCountries()
+
+    const lastIP = userIP.lastIPs.sort(
+      (a, b) => b.lastConnection.getTime() - a.lastConnection.getTime(),
+    )[0]
+
+    if (!lastIP) return true
+
+    if (lastIP.proxy && lastIP.proxy == "yes") return new InvalidIpMetadataTypeError()
```

Comment:
the statement `lastIP.proxy == "yes"` feels a bit off. if proxy is a boolean, we should type it that way, and convert from a string to a boolean at the service level.